Build fails on Ubuntu 16.04 LTS due to missing net_sockets.h in libmbedtls-dev

Hi, I noticed that a [recent mbedtls commit](https://github.com/libevent/libevent/commit/b28effa95082e468340cc13d274d3a2ea205c174) broke building libevent from git master on Ubuntu 16.04 LTS:


```
$ git clone https://github.com/libevent/libevent.git
$ cd libevent
$ ./autogen.sh
$ ./configure
$ make
[...]
  CC       libevent_mbedtls_la-bufferevent_mbedtls.lo
bufferevent_mbedtls.c:28:10: fatal error: mbedtls/net_sockets.h: No such file or directory
 #include <mbedtls/net_sockets.h>
                 ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [libevent_mbedtls_la-bufferevent_mbedtls.lo] Error 1
make: *** [all] Error 2
Makefile:1926: recipe for target 'libevent_mbedtls_la-bufferevent_mbedtls.lo' failed
make[1]: Leaving directory '/tmp/bz/libevent'
Makefile:1411: recipe for target 'all' failed
make  install-am
make[1]: Entering directory '/tmp/bz/libevent'
  CC       libevent_mbedtls_la-bufferevent_mbedtls.lo
bufferevent_mbedtls.c:28:10: fatal error: mbedtls/net_sockets.h: No such file or directory
 #include <mbedtls/net_sockets.h>
                 ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Makefile:1926: recipe for target 'libevent_mbedtls_la-bufferevent_mbedtls.lo' failed
make[1]: *** [libevent_mbedtls_la-bufferevent_mbedtls.lo] Error 1
make[1]: Leaving directory '/tmp/bz/libevent'
Makefile:2933: recipe for target 'install' failed
make: *** [install] Error 2
```

```
$ apt-cache policy libmbedtls-dev
libmbedtls-dev:
  Installed: 2.2.1-2ubuntu0.3
  Candidate: 2.2.1-2ubuntu0.3
  Version table:
 *** 2.2.1-2ubuntu0.3 500
        500 http://mirrors.digitalocean.com/ubuntu xenial-updates/universe amd64 Packages
        500 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages
        100 /var/lib/dpkg/status
     2.2.1-2 500
        500 http://mirrors.digitalocean.com/ubuntu xenial/universe amd64 Packages
```

Looking at the [libmbedtls-dev package file list](https://packages.ubuntu.com/xenial/amd64/libmbedtls-dev/filelist) confirms the Xenial version does _not_ contain `net_sockets.h`. For now I have used the --disable-mbedtls flag as a workaround. Considering 16.04 is still officially supported, I wonder if/where this should be reported. Just wanted to file this FYI here so users are aware of it.

>bufferevent_mbedtls.c:28:10: fatal error: mbedtls/net_sockets.h: No such file or directory
> #include <mbedtls/net_sockets.h>
>                 ^~~~~~~~~~~~~~~~~~~~~~~

FWIW I saw this issues on CI too

--- a/Makefile.am
+++ b/Makefile.am
@@ -337,6 +337,7 @@ noinst_HEADERS +=				\
 	time-internal.h				\
 	util-internal.h				\
 	openssl-compat.h			\
+	mbedtls-compat.h			\
 	ssl-compat.h			    \
 	wepoll.h
 

--- a/bufferevent_mbedtls.c
+++ b/bufferevent_mbedtls.c
@@ -24,8 +24,8 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "mbedtls-compat.h"
 #include <mbedtls/ssl.h>
-#include <mbedtls/net_sockets.h>
 #include <mbedtls/error.h>
 
 #include "event2/util.h"

--- a/mbedtls-compat.h
+++ b/mbedtls-compat.h
@@ -1,0 +1,11 @@
+#ifndef MBEDTLS_COMPAT_H
+#define MBEDTLS_COMPAT_H
+
+#include <mbedtls/version.h>
+#if MBEDTLS_VERSION_MAJOR < 2 || (MBEDTLS_VERSION_MAJOR == 2 && MBEDTLS_VERSION_MINOR < 4)
+#include <mbedtls/net.h>
+#else
+#include <mbedtls/net_sockets.h>
+#endif
+
+#endif // LIBEVENT_MBEDTLS_COMPAT_H

--- a/sample/ssl-client-mbedtls.c
+++ b/sample/ssl-client-mbedtls.c
@@ -22,7 +22,7 @@
 #include "mbedtls/config.h"
 #include "mbedtls/platform.h"
 
-#include "mbedtls/net_sockets.h"
+#include "mbedtls-compat.h"
 #include "mbedtls/debug.h"
 #include "mbedtls/ssl.h"
 #include "mbedtls/entropy.h"

